crates/accelerate/src/uc_gate.rs

```diff
@@ -119,8 +128,8 @@ pub fn dec_ucg_help(
     let mut single_qubit_gates: Vec<Matrix2<Complex64>> = sq_gates
         .into_iter()
         .map(|x| {
-            let res: MatrixView2<Complex64> = x.try_as_matrix().unwrap();
-            res.into_owned()
+            let arr = x.as_array();
+            Matrix2::new(arr[[0, 0]], arr[[0, 1]], arr[[1, 0]], arr[[1, 1]])
```

Comment:
Why was this change needed? Was something failing with it using the `try_as_matrix()` function? I know it would fail if you gave it something other than a 2x2 was it failing also if the array was row ordered too or something? My concern with this is there is no check that the input here is a 2x2. So you could pass this function a much larger array and nothing would know.

That being said it's not critical because this is in an internal function and if something gave it not a 2x2 that's a bug elsewhere in the code. I'm just curious what motivated this change.

Reply:
Yes, the `.try_as_matrix().unwrap()` was failing; I don't really understand why.

Reply:
Do you remember the error message or anything? I have a feeling it was either the column order or something. But if there is an example we could probably fix it python side